lntypes/channel_party.go

```diff
@@ -0,0 +1,21 @@
+package lntypes
+
+// ChannelParty is a type used to have an unambiguous description of which node
+// is being referred to. This eliminates the need to describe as "local" or
+// "remote" using bool.
+type ChannelParty byte
+
+const (
+	// Local is a ChannelParty constructor that is used to refer to the
+	// node that is running.
+	Local ChannelParty = iota
+
+	// Remote is a ChannelParty constructor that is used to refer to the
+	// node on the other end of the peer connection.
+	Remote
+)
+
+// Not inverts the role of the ChannelParty.
+func (p ChannelParty) CounterParty() ChannelParty {
+	return ChannelParty((byte(p) + 1) % 2)
```

Comment:
At the cost of few more lines, could break this out into a switch/else statement, to make it more obvious what's going on at a glance. Or use some look up map like: `opposite`: 
```go
func (p ChannelParty) CounterParty() ChannelParty {
    switch p {
      case Local:
         return Remote
      case Remote:
         return Local
    }
}
```

---
```go
var inverseChannelParty = map[ChannelParty]ChannelParty{
    Local: Remote,
    Remote: Local, 
}

func (p ChannelParty) CounterParty() ChannelParty {
    return inverseChannelParty[p]
}
```

Reply:
idk what the perf characteristics of the map are but I can def go for the switch case construction.